### PR TITLE
full refactor of proof objects and registration

### PR DESCRIPTION
**Pull request by maxtaco**
- more getting us ready to factor our the proof classes
- now we're pretty close, just have to move files over and reshuffle
  objects dangling off of G.

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @patrickxb 

More refactoring in preparation for pulling out proofs from libkb.....

File: go/libkb/services.go [hunk 1]
```diff
@@ -122,12 +153,14 @@ func (t BaseServiceType) GetAPIArgKey() string {
 	return "remote_username"
 }
 
+func (t BaseServiceType) IsDevelOnly() bool { return false }
```
Comment (patrickxb) [inline go/libkb/services.go]: Doesn't rooter need to define this method and return true?

Comment (maxtaco) [inline go/libkb/services.go]: urg, i thought it did...

Comment (maxtaco) [inline go/libkb/services.go]: thanks, should be fixed

Comment (maxtaco) [inline go/libkb/services.go]: I added this fix upstream in CORE-3574-3

**Comment by patrickxb** [comment]
LGTM

**Pull request merged**